packages/material-ui-lab/src/SliderStyled/SliderStyled.js

```diff
@@ -0,0 +1,353 @@
+import * as React from 'react';
+import PropTypes from 'prop-types';
+import { useThemeProps, muiStyled, fade, lighten, darken } from '@material-ui/core/styles';
+import { propsToClassKey } from '@material-ui/styles';
+import { capitalize } from '@material-ui/core/utils';
+import SliderUnstyled from '../SliderUnstyled';
+
+const overridesResolver = (props, styles, name) => {
+  const {
+    color = 'primary',
+    marks: marksProp = false,
+    max = 100,
+    min = 0,
+    orientation = 'horizontal',
+    step = 1,
+    track = 'normal',
+  } = props;
+
+  const marks =
+    marksProp === true && step !== null
+      ? [...Array(Math.floor((max - min) / step) + 1)].map((_, index) => ({
+          value: min + step * index,
+        }))
+      : marksProp || [];
+
+  const marked = marks.length > 0 && marks.some((mark) => mark.label);
+
+  const styleOverrides = {
+    ...styles.root,
+    ...styles[`color${capitalize(color)}`],
+    '&.Mui-disabled': styles.disabled,
+    ...(marked && styles.marked),
+    ...(orientation === 'vertical' && styles.vertical),
+    ...(track === 'inverted' && styles.trackInverted),
+    ...(track === false && styles.trackFalse),
+    [`.& ${name}-rail`]: styles.rail,
+    [`.& ${name}-track`]: styles.track,
+    [`.& ${name}-mark`]: styles.mark,
+    [`.& ${name}-markLabel`]: styles.markLabel,
+    [`.& ${name}-valueLabel`]: styles.valueLabel,
+    [`.& ${name}-thumb`]: {
+      ...styles.thumb,
+      ...styles[`thumbColor${capitalize(color)}`],
+      '&.Mui-disabled': styles.disabled,
+    },
+  };
+
+  return styleOverrides;
+};
+
+const variantsResolver = (props, styles, theme, name) => {
+  const { state = {} } = props;
+  let variantsStyles = {};
+  if (theme && theme.components && theme.components[name] && theme.components[name].variants) {
+    const themeVariants = theme.components[name].variants;
+    themeVariants.forEach((themeVariant) => {
+      let isMatch = true;
+      Object.keys(themeVariant.props).forEach((key) => {
+        if (state[key] !== themeVariant.props[key]) {
+          isMatch = false;
+        }
+      });
+      if (isMatch) {
+        variantsStyles = { ...variantsStyles, ...styles[propsToClassKey(themeVariant.props)] };
+      }
+    });
+  }
+
+  return variantsStyles;
+};
+
+const SliderRoot = muiStyled(
+  'span',
+  {},
+  { muiName: 'MuiSlider', overridesResolver, variantsResolver },
+)((props) => {
+  return {
+    height: 2,
+    width: '100%',
+    boxSizing: 'content-box',
+    padding: '13px 0',
+    display: 'inline-block',
+    position: 'relative',
+    cursor: 'pointer',
+    touchAction: 'none',
+    color: props.theme.palette.primary.main,
+    WebkitTapHighlightColor: 'transparent',
+    ...(props.state.color === 'secondary' && {
```

Comment:
Why is `color` a state?

Reply:
All style related props which we want to use in the `styled` utility are under `props.state` (should we pick up better name?). The main reason for this, is to have a common `shouldSpreadProp` logic, where we are just not spreading the `state` to the DOM element. Otherwise we will need to have custom logic for each component, and re-eval everytime a new prop is added.

Reply:
> All style related props which we want to use in the styled utility are under props.state (should we pick up better name?).

`props.style` is kind of what you're describing.

Reply:
style is already a valid prop, maybe styleProps?

Reply:
Renamed to `state` to `styleProps`